bazel/envoy_build_system.bzl

```diff
@@ -75,6 +75,9 @@ def envoy_copts(repository, test = False):
            }) + select({
                repository + "//bazel:disable_signal_trace": [],
                "//conditions:default": ["-DENVOY_HANDLE_SIGNALS"],
+           }) + select({
+               repository + "//bazel:optimize_bin_size": ["-Os"],
```

Comment:
Did you compare this vs passing `--copt=-Os` to bazel? `envoy_copts` doesn't apply to dependency but CLI copt is. So I think `--copt=-Os` may produce smaller binary.

Reply:
+1, we need to compile all dependencies with -Os also. Also, can we make sure we are compiling with symbols also? They can be stripped after.